### PR TITLE
GPUComputationRenderer: Added check to make it work on iOS.

### LINKED ISSUE #9628: GPGPU examples do not work on iOS Safari or Chrome
##### Description of the problem 

```
[Warning] WebGL: INVALID_FRAMEBUFFER_OPERATION: clear: framebuffer not complete (three.js, line 19838)
[Warning] WebGL: INVALID_FRAMEBUFFER_OPERATION: drawElements: framebuffer not complete (three.js, line 16994)
```

This is happening in the GPGPU examples and in my own project (see [joncole.me/reaction-diffusion](http://www.joncole.me/reaction-diffusion)), which uses the latest build, indicating that whatever problem this is spans multiple versions of Three. It seems to me that the problem is related to the use of floating-point textures, but my knowledge ends there. 

The problem is that the things the shaders do in these examples (i.e. forming the waves in the water example) don't seem to take, and it will repeatedly spit out the two errors above.

I could reproduce this on an iPad (9.3.3), iPhone 6 (9.0.2), and iPhone 6s (9.3.4), using both Chrome and Safari. I do not know if this affects Android, but it seems likely.

##### Three.js version

- [x] Dev
- [x] r80
- [ ] ...

##### Browser

- [x] All of them
- [ ] Chrome
- [ ] Firefox
- [ ] Internet Explorer

##### OS

- [ ] All of them
- [ ] Windows
- [ ] Linux
- [ ] Android
- [x] IOS

##### Hardware Requirements (graphics card, VR Device, ...)
Any device with iOS 8.0+, most likely

### PR DESCRIPTION
**Pull request by yomboprime**
Fixes #9628 
It checks if FloatType is supported by trying to render in a temporary render target.
If the check fails, it switches to HalfFloatType which works on iOS.

Tested by @seieibob on an iPhone and an iPad.

### CODE REVIEW

File: examples/js/GPUComputationRenderer.js [hunk 1]
```diff
@@ -291,13 +302,15 @@ function GPUComputationRenderer( sizeX, sizeY, renderer ) {
...
 
+		textureType = textureType || this.defaultTextureType;
+
 		var renderTarget = new THREE.WebGLRenderTarget( sizeXTexture, sizeYTexture, {
 			wrapS: wrapS,
 			wrapT: wrapT,
 			minFilter: minFilter,
 			magFilter: magFilter,
 			format: THREE.RGBAFormat,
-			type: THREE.FloatType,
```
Comment (mrdoob) [inline examples/js/GPUComputationRenderer.js]: I think I would keep things a bit simpler for now and just change this line to this:

```js
type: ( /(iPad|iPhone|iPod)/g.test( navigator.userAgent ) ) ? THREE.HalfFloatType : THREE.FloatType,
```

**Comment by yomboprime** [comment]
Okay, I did the change with the regular expression.

**Comment by mrdoob** [comment]
Thanks!

**Comment by yomboprime** [comment]
;-)

**Pull request merged**